### PR TITLE
Add UI for managing Friend Requests

### PR DESCRIPTION
**Pull request by lunasrc**
This pull request introduces the functionality for users to link with others in the app, enhancing the social connectivity experience. Users can now perform actions such as sending requests, canceling requests, accepting requests, or unfriending other users.
### Search Screen: 
A dynamic button allows users to:
- Send a friend request to the displayed user.
- Cancel a previously sent request.
- Accept an incoming friend request.
- Unfriend a currently linked user.
### Notifications Screen:
- Users can now view and manage their sent requests and received requests from a centralized location.
- Sent requests are displayed in a dedicated section, allowing users to review and cancel them if needed.
- Received requests are also displayed, enabling users to quickly accept them.
### Profile Screen:
- Clicking on the Links now opens a screen with the list of friends the current user is linked with.

Screenshot 2024-12-12 at 14 23 39

Screenshot 2024-12-12 at 14 24 16

### CODE REVIEW

File: app/src/main/java/com/epfl/beatlink/viewmodel/profile/FriendRequestViewModel.kt [hunk 1]
```diff
@@ -158,4 +165,17 @@ open class FriendRequestViewModel(
       }
     }
   }
+
```
Comment (sandroquist) [inline app/src/main/java/com/epfl/beatlink/viewmodel/profile/FriendRequestViewModel.kt:168]: Don't forget to give a little bit of documentation when adding functions

File: app/src/main/java/com/epfl/beatlink/viewmodel/profile/ProfileViewModel.kt [hunk 2]
```diff
@@ -89,7 +110,12 @@ open class ProfileViewModel(
     _isProfileUpdated.value = false
   }
 
-  fun getUsername(userId: String, onResult: (String?) -> Unit) {
+  fun clearSelectedUser() {
```
Comment (sandroquist) [inline app/src/main/java/com/epfl/beatlink/viewmodel/profile/ProfileViewModel.kt:113]: Here as well, don't forget to give documentation for new functions (there are multiple in this file)

**Review by sandroquist — CHANGES_REQUESTED** [review]
Your pr is clean, well structured and the feature works well. I suggested a few quick changes to improve readability and also, don't forget to document when adding new functions.

File: app/src/main/java/com/epfl/beatlink/ui/components/profile/ProfileColumn.kt [hunk 3]
```diff
@@ -19,65 +17,111 @@ import androidx.compose.foundation.lazy.LazyColumn
...
- * @param topSongsState A list of [SpotifyTrack] objects representing the user's top songs.
- * @param topArtistsState A list of [SpotifyArtist] objects representing the user's top artists.
- * @param userPlaylists A list of [UserPlaylist] objects representing the user's playlists.
- * @param paddingValue Padding values to be applied to the column layout.
- * @param profilePicture A mutable state containing the user's profile picture as a [Bitmap].
- * @param ownProfile A boolean variable that determines wich type of Profile screen needs to be
- *   displayed.
- * @param buttonTestTag A test tag for a button that changes depending on the Profile screen
- *   displayed
  */
```
[the remark this answers is not in the document]
Comment (lunasrc) [reply app/src/main/java/com/epfl/beatlink/ui/components/profile/ProfileColumn.kt:56]: I changed the parameters but I will update it

**Review by sandroquist — APPROVED** [review]
Great job on addressing the requested changes! The updates are clear and well-implemented. It's ready to be merged!

**Pull request merged**